cmd/snap/cmd_run.go

```diff
@@ -67,7 +67,7 @@ func (x *cmdRun) Execute(args []string) error {
 	if x.Hook != "" && x.Command != "" {
 		return fmt.Errorf("cannot use --hook and --command together")
 	}
-	if x.Revision != "" && x.Hook == "" {
+	if x.Revision != "unset" && x.Revision != "" && x.Hook == "" {
```

Comment:
Indeed, I don't envision anyone actually typing `-r=unset`, but it simplified a bit of the code. See [this discussion](https://github.com/snapcore/snapd/pull/1511#discussion-diff-70055516) for some background, and let me know if you'd like it changed back.